lib/expr.ml

```diff
@@ -515,41 +513,45 @@ module Bool = struct
     | Val False -> Some false
     | _ -> None
 
-  let v b = make (match b with true -> Val True | false -> Val False)
+  let true_ = value True
+
+  let false_ = value False
+
+  let to_val b = if b then true_ else false_
 
-  let not (b : t) =
+  let not_ (b : t) =
     match of_val (view b) with
-    | Some b -> v (not b)
+    | Some b -> to_val b
     | None -> (
       match view b with
       | Unop (Ty_bool, Not, cond) -> cond
       | _ -> unop' Ty_bool Not b )
 
-  let ( = ) (b1 : t) (b2 : t) =
+  let equal (b1 : t) (b2 : t) =
     match (view b1, view b2) with
-    | Val True, Val True | Val False, Val False -> value True
-    | _ -> relop' Ty_bool Eq b1 b2
+    | Val True, Val True | Val False, Val False -> true_
+    | _ -> relop Ty_bool Eq b1 b2
 
   let distinct (b1 : t) (b2 : t) =
     match (view b1, view b2) with
-    | Val True, Val False | Val False, Val True -> value True
-    | _ -> relop' Ty_bool Ne b1 b2
+    | Val True, Val False | Val False, Val True -> true_
+    | _ -> relop Ty_bool Ne b1 b2
 
   let and_ (b1 : t) (b2 : t) =
     match (of_val (view b1), of_val (view b2)) with
-    | Some b1, Some b2 -> v (b1 && b2)
+    | Some b1, Some b2 -> to_val (b1 && b2)
     | Some true, _ -> b2
     | _, Some true -> b1
```

Comment:
You can merge the two cases:

`| Some true, b | b, Some true -> b`

Reply:
Unfortunately, I cannot merge this. `b` would have type `bool option` instead of `Expr.t`